miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppManifestCache.kt

```diff
@@ -0,0 +1,64 @@
+package com.rakuten.tech.mobile.miniapp.storage
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.rakuten.tech.mobile.miniapp.MiniAppManifest
+import java.lang.Exception
+
+/**
+ * A caching class to read and store the [MiniAppManifest] per MiniApp
+ * using [SharedPreferences].
+ */
+// TODO: unit testcases
+internal class MiniAppManifestCache(context: Context) {
+    private val prefs: SharedPreferences = context.getSharedPreferences(
+        "com.rakuten.tech.mobile.miniapp.manifest.cache", Context.MODE_PRIVATE
+    )
+
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun doesDataExist(miniAppId: String) = prefs.contains(miniAppId)
+
+    /**
+     * Reads the grant results from SharedPreferences.
+     * @param [miniAppId] the key provided to find the stored manifest per MiniApp.
+     * @return [MiniAppManifest] an object to contain the manifest per MiniApp.
+     * if data has been stored in cache, otherwise empty value.
+     */
+    @SuppressLint("RestrictedApi")
```

Comment:
See my comment above: I think you don't need this if you remove the `otherwise`.